docs/contributor/updateability.md

```diff
@@ -405,9 +405,59 @@ file is created with version of the executable.
 If file existed and version checks were successful, the file is updated 
 with version of the executable.
 
-### Migration
+### Kubernetes Storage Migration
 
-TODO
+Storage migration is process of updating objects to newer versions,
+for example from `v1beta1` to `v1beta2`.
+
+Initially, the plan was to perform the migration with only etcd and 
+kube-apiserver running.
+This turned out to not be a viable options because migrating
+CustomResourceDefinitions might require webhooks which run as a Pods, 
+therefore requiring kubelet, CNI, DNS, scheduler, and other components.
+
+Right now, storage migration is comprised of two additional controllers
+embedded into the MicroShift: Kube Storage Version Trigger and
+Kube Storage Version Migrator. They make use of a `StorageVersionMigration` CR.
+
+### `StorageVersionMigration` Custom Resource
+
+It is a Custom Resources created by the Trigger and
+picked up to handle by the Migrator.
+It can be also used to monitor progress of the migration and get
+information about any failures.
+
+#### Kube Storage Version Trigger
+
+Trigger performs a discovery which is a process of gathering information about
+existing Kubernetes Resources and their versions.
+If the Trigger controller discovers that the Resource should be migrated,
+a `StorageVersionMigration` CR is created which is then picked up by 
+the Migrator.
+
+Trigger controller performs the discovery and creates `StorageVersionMigration`
```

Comment:
A bit more context here with this.

> Trigger controller performs the discovery and creates a `StorageState` and a `StorageVersionMigration` respectively. The `StorageState` contains the current discovered state of the API resource and the previous, if there is a version difference for that specific API resource, then a `StorageVersionMigration` is created by the trigger. This check happens on start and every 10 minutes (discovery period; hardcoded in upstream code).

Reply:
Thanks! I haven't noticed that one when skimming through the code